### DISCUSSION TITLE
Command to run attach().

### DISCUSSION THREAD

**Discussion by JoseConseco** [mfussenegger/nvim-dap#386] [Q&A] [answered]
In past we could use attach(host, port, config)  : 
![image](https://user-images.githubusercontent.com/13521338/146680823-716ab420-5a6b-4e8d-a11e-665f13fbdcc2.png)

But now I'm not sure how do I pass the adapter and config arguments ?  My objective is to  attach to debugpy:   localhost, 5678,  without constantly having to go through setup settings:
![image](https://user-images.githubusercontent.com/13521338/146680886-2772c8bb-552f-4e4d-b7fa-46cae1ebe4ee.png)

I tried something like: `lua require'dap'.attach(require'dap'.adapter.python={request='attach','0.0.0.0', 5678},{})`
but it does not seem to work...

**Comment by mfussenegger** [comment, marked as answer]
You could call `dap.run()` directly if you want to skip the configurations prompt. E.g. `dap.run(dap.configurations[vim.bo.filetype][3])` if you know the configuration entry you want is on position 3.

Another option is to use `dap.continue()` the first time and then for sub-sequent debug sessions use `dap.run_last()` to skip the prompt - it will immediately re-use the last configuration.

**Comment by JoseConseco** [comment]
Thanks to your answer I manage to get this:
`:lua require'dap'.run({type='python', request='attach', host='127.0.0.1', port=5678})`
So finally I do not have to enter host and port every time   (or press ret 3x - to use default values)

**Reply by mfussenegger** [reply]
If you always use the same host and port number you could also just add a configuration that has fixed host/port entries instead of a configuration that uses the input function.

The configurations are meant to be customized/user specific.
